promql/engine.go

```diff
@@ -2553,12 +2553,20 @@ func vectorElemBinop(op parser.ItemType, lhs, rhs float64, hlhs, hrhs *histogram
 	switch op {
 	case parser.ADD:
 		if hlhs != nil && hrhs != nil {
-			return 0, hlhs.Copy().Add(hrhs).Compact(0), true
+			res, err := hlhs.Copy().Add(hrhs)
+			if err != nil {
+				panic(err)
```

Comment:
We should not panic in the engine, instead return false in the last return value (keep).
```suggestion
				return 0, nil, false
```